### ISSUE TITLE
During DDP training timm densenet121, mobilenetv2(v3) models do not save state_dict correctly.

### ISSUE THREAD

**Issue by linchundan88** [pytorch/pytorch#83245]
### 🐛 Describe the bug

During DDP training timm densenet121, mobilenetv2(v3) models do not save state_dict correctly.
 After load the stat dicts, the model do not act well.  However, when using torch.save save the model, it works well.

### Versions

both 1.11, 1.12 and 1.12.1

**Comment by linchundan88** [comment]
However, other models such timm resnet50, inception-v3,xception, etc work well with DDP torch.save with save_dicts.
I do not know it is related to timm or pytorch.

**Comment by linchundan88** [comment]
if rank == 0:
            save_model_file = save_model_dir / f'valid_loss_{loss_valid_avg:.4f}_epoch{epoch}.pth'  # the loss of the last validation dataset
            if not torch_script:
                print('save model:', save_model_file)
                # torch.save(model_ddp.module.state_dict(), save_model_file) #model_ddp.state_dict() model_ddp.module.state_dict()
                torch.save(model_ddp.module, save_model_file)

   
  If I use torch.save(model_ddp.module.state_dict(), save_model_file),  when loading the state dict, the model can not give right predicted results. Howeven when using torch.save save the model, it is OK.
  It only happend for mobilenet, densenet121, and cspresnet models, other models such as resnet, inception, Vit, swin transformer, etc, worked fine.

**Comment by rohan-varma** [comment]
Hi, thanks for reporting the issue! A couple of q's:

1. Does the issue go away without DDP, i.e. local only training? 
2. Is there a difference if you use `ddp.state_dict()` directly (you'll have to strip out the module prefix when loading the model back) versus `ddp.module.state_dict()`? 
3. You mentioned this only occurs for some models - would it be possible to get an e2e repro of this?
